lib/rules/no-unused-vars.js

```diff
@@ -858,10 +861,357 @@ module.exports = {
                             messageId: "unusedVar",
                             data: unusedVar.references.some(ref => ref.isWrite())
                                 ? getAssignedMessageData(unusedVar)
-                                : getDefinedMessageData(unusedVar)
+                                : getDefinedMessageData(unusedVar),
+                            suggest: [
+                                {
+                                    messageId: "removeVar",
+                                    data: {
+                                        varName: unusedVar.name
+                                    },
+                                    fix(fixer) {
+                                        const id = unusedVar.identifiers[0];
+                                        const parent = id.parent;
+                                        const tokenBefore = sourceCode.getTokenBefore(id);
+                                        const tokenAfter = sourceCode.getTokenAfter(id);
+
+                                        /**
+                                         * Gets fixes in variable declarations and function parameters
+                                         * @param {ASTNode} node parent node of identifier
+                                         * @returns {Object} fixer object
+                                         */
+                                        function fixVariables(node) {
+                                            if (node.parent.type === "VariableDeclarator") {
+                                                if (node.parent.parent.declarations.length === 1) {
+                                                    return fixer.removeRange(node.parent.parent.range);
+                                                }
+
+                                                if (sourceCode.getTokenBefore(node.parent).value === ",") {
+                                                    return fixer.removeRange([sourceCode.getTokenBefore(node.parent).range[0], node.parent.range[1]]);
+                                                }
+
+                                                return fixer.removeRange([node.parent.range[0], sourceCode.getTokenAfter(node.parent).range[1]]);
+                                            }
+
+                                            if (node.parent.type === "FunctionDeclaration" || node.parent.type === "FunctionExpression") {
+                                                if (node.parent.params.length === 1) {
+                                                    return fixer.removeRange(node.range);
+                                                }
+
+                                                if (sourceCode.getTokenBefore(node).value === "(" && sourceCode.getTokenAfter(node).value === ",") {
```

Comment:
You're also doing `getTokenBefore(node)` and `getTokenAfter(node)` a few times here. It's best to store those values in a variable to save the function call.

Reply:
Done!